src/rime/gear/simplifier.cc

```diff
@@ -46,19 +46,46 @@ class Opencc {
   }
 
   bool ConvertWord(const string& text, vector<string>* forms) {
-    if (dict_ == nullptr)
-      return false;
-    opencc::Optional<const opencc::DictEntry*> item = dict_->Match(text);
-    if (item.IsNull()) {
-      // Match not found
+    if (converter_ == nullptr) {
       return false;
-    } else {
-      const opencc::DictEntry* entry = item.Get();
-      for (auto&& value : entry->Values()) {
-        forms->push_back(std::move(value));
+    }
+    const list<opencc::ConversionPtr> conversions =
+        converter_->GetConversionChain()->GetConversions();
+    vector<string> original_words{text};
+    bool matched = false;
+    for (auto conversion : conversions) {
+      opencc::DictPtr dict = conversion->GetDict();
+      if (dict == nullptr) {
+        return false;
       }
-      return forms->size() > 0;
+      set<string> word_set;
+      vector<string> converted_words;
+      for (const auto& original_word : original_words) {
+        opencc::Optional<const opencc::DictEntry*> item =
+            dict->Match(original_word);
+        if (item.IsNull()) {
+          // Current dictionary doesn't convert the word. We need to keep it for
+          // other dicts in the chain. e.g. s2t.json expands 里 to 里 and 裏,
+          // then t2tw.json passes 里 as-is and converts 裏 to 裡.
+          converted_words.push_back(original_word);
```

Comment:
Maybe I'm nitpicking, but does it need de-duplication here too?

I cannot find a real testcase creating duplication words in Chinese conversions. So It's OK to push back directly.

Reply:
Thanks for the catch! I don't want to leave a potential bug here.